sql/log_event.cc

```diff
@@ -11845,6 +11845,23 @@ int Rows_log_event::do_apply_event(Relay_log_info const *rli)
     }
   } // if (table)
 
+#ifdef WITH_WSREP
+  if (WSREP(thd) && error == HA_ERR_RECORD_FILE_FULL)
+  {
+    /*
+      Prevent a node to be aborted to make it easier for an administrator
+      to solve the problem later.
+
+      Aborting the commit is required to release Galera monitor from the
+      transaction.
+    */
+    wsrep->desync(wsrep);
```

Comment:
> Desync only turns off the flow controll, applying still takes place, so I assume the result of the current implementation will be constant error logs that commit failed because of out-of-space.

True

> Is it what we need or rather we want to pause writeset applying?

I don't know, I assumed that's very rare situation and log spamming can be acceptable. But pause probably is more accurate.

> It would be nice to test it in MTR
> Additionally: maybe worth adding explicit message/error log

Yes, I think it's good ideas, I'll do this.